src/lib/Imp.hs

```diff
@@ -420,30 +420,17 @@ offsetTo ty i     = fromScalarAtom <$> fromEmbed (offsetToE ty (toScalarAtom Int
 elemCountE :: MonadEmbed m => ScalarTableType -> m Atom
 elemCountE ty = case ty of
   BaseTy _  -> return $ IntVal 1
-  TabTy (NoName:>idxTy) b -> bindM2 imul (indexSetSizeE idxTy) (elemCountE b)
-  TabTy v (TabTy (NoName :> TC (IndexRange _ Unlimited (InclusiveLim (Var v')))) (BaseTy _)) | v == v' ->
-    triangularSum =<< indexSetSizeE (varType v)
-  TabTy _ _ ->
-    error $ "Tables with sizes of dimensions dependent on previous dimensions are not supported: " ++ pprint ty
+  TabTy v _ -> offsetToE ty =<< indexSetSizeE (varType v)
   _ -> error $ "Not a scalar table type: " ++ pprint ty
 
 -- TODO: Accept an index instead of an ordinal?
 offsetToE :: MonadEmbed m => ScalarTableType -> Atom -> m Atom
 offsetToE ty i = case ty of
-  TabTy (NoName:>_) b -> imul i =<< elemCountE b
-  TabTy v (TabTy (NoName :> TC (IndexRange _ Unlimited (InclusiveLim (Var v')))) (BaseTy _)) | v == v' ->
-    triangularSum i
-  TabTy _ _ ->
-    error $ "Tables with sizes of dimensions dependent on previous dimensions are not supported: " ++ pprint ty
   BaseTy _  -> error "Indexing into a scalar!"
+  TabTy (NoName :> _) b -> imul i =<< elemCountE b
```

Comment:
This is only necessary, because the algebra backend cannot handle exclusive bounds in index ranges. It should be removed soon!